packages/cli/src/util/config.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { JSONSchemaType } from "ajv";
+import ajvModule from "ajv";
+import { existsSync, promises as fsPromises } from "node:fs";
+import path from "node:path";
+
+export interface SiteConfig {
+  application: string;
+  directory: string;
+  autoVersion?: boolean;
+}
+
+export interface FoundryConfig {
+  foundryUrl: string;
+  site: SiteConfig;
+}
+
+export interface LoadedFoundryConfig {
+  foundryConfig: FoundryConfig;
+  configFilePath: string;
+}
+
+const CONFIG_FILE_NAMES: string[] = [
+  "foundry.config.json",
+];
+
+const configFileSchema: JSONSchemaType<FoundryConfig> = {
+  type: "object",
+  properties: {
+    foundryUrl: { type: "string" },
+    site: {
+      type: "object",
+      properties: {
+        application: { type: "string" },
+        directory: { type: "string" },
+        autoVersion: { type: "boolean", nullable: true },
+      },
+      required: ["application", "directory"],
+    },
+  },
+  required: ["foundryUrl", "site"],
+  additionalProperties: false,
+};
+
+const Ajv = ajvModule.default; // https://github.com/ajv-validator/ajv/issues/2132
+const ajv = new Ajv({ allErrors: true });
+const validate = ajv.compile(configFileSchema);
```

Comment:
Given that `loadFoundryConfig()` will only be called once per lifetime of the command it may be better to construct this when needed lazily with dynamic import rather than immediately once the script is loaded